### PR TITLE
cleanup sandbox on start of deployment processing

### PR DESCRIPTION
**Pull request by jw-msft**

### CODE REVIEW

**Review by nihemstr — APPROVED** [review]

File: src/utils/system_utils/src/system_utils.c [hunk 1]
```diff
@@ -739,3 +740,75 @@ _Bool SystemUtils_IsFile(const char* path, int* err)
     }
     return is_file;
 }
+
+/**
+ * @brief For each dir_name in the baseDir, it calls the action function with the baseDir and dir_name.
+ * @param baseDir The base dir to list directories in.
+ * @param excludedDir A dir to exclude via exact match in addition to . and .. dirs. NULL means to exclude only . and .. dirs.
+ * @param perDirActionFunctor The functor to apply for each dir in the baseDir.
+ * @param err the error code (optional, can be NULL)
```
Comment (Nox-MSFT) [inline src/utils/system_utils/src/system_utils.c:749]: Remove. 'err' is not a param.

File: src/utils/system_utils/src/system_utils.c [hunk 2]
```diff
@@ -739,3 +740,75 @@ _Bool SystemUtils_IsFile(const char* path, int* err)
...
+    {
+        errno = 0;
+        if ((dir_entry = readdir(dir)) != NULL)
+        {
+            if (strcmp(dir_entry->d_name, ".") == 0 || strcmp(dir_entry->d_name, "..") == 0)
+            {
+                continue;
+            }
+
+            if (excludedDir != NULL && strcmp(dir_entry->d_name, excludedDir) == 0)
```
Comment (Nox-MSFT) [inline src/utils/system_utils/src/system_utils.c:782]: nit: can combine line 777 & 782

File: src/utils/system_utils/src/system_utils.c [hunk 3]
```diff
@@ -739,3 +740,75 @@ _Bool SystemUtils_IsFile(const char* path, int* err)
...
+        }
+        else
+        {
+            if (errno != 0)
+            {
+                // error
+                err_ret = errno;
+                goto done;
+            }
+            else
```
Comment (Nox-MSFT) [inline src/utils/system_utils/src/system_utils.c:797]: Do we need this 'else' block? You already goto done at line 794 anyway.

File: src/adu_workflow/src/agent_workflow.c [hunk 4]
```diff
@@ -102,16 +102,87 @@ static const char* ADUCITF_WorkflowStepToString(ADUCITF_WorkflowStep workflowSte
...
+
+    *lastSlash = '\0';
+
+    if (!SystemUtils_IsDir(workFolder, &err) || err != 0)
+    {
+        Log_Error("%s is not a dir", workFolder);
+        goto done;
+    }
+
+    Log_Debug("Cleaning foreach under %s except %s", workFolder, current_workflowId);
```
Comment (Nox-MSFT) [inline src/adu_workflow/src/agent_workflow.c:173]: "Cleaning **dirs** under %s..."

**Review by Nox-MSFT — APPROVED** [review]
Approved, with minor comments.

File: src/utils/system_utils/src/system_utils.c (hunk 2 above)
Comment (jw-msft) [reply src/utils/system_utils/src/system_utils.c:782]: leaving as it's easier to debug by keeping separate

File: src/utils/system_utils/src/system_utils.c (hunk 3 above)
Comment (jw-msft) [reply src/utils/system_utils/src/system_utils.c:797]: removed but left some comment

**Pull request merged**